In [6]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [8]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/A232/Data Analytics/lab/data.csv')

In [12]:
# printing first 10 rows of the DataFrame
df.head(10)

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.0
6        60    110       136     374.0
7        45    104       134     253.3
8        30    109       133     195.1
9        60     98       124     269.0

In [13]:
# printing last 5 rows of the DataFrame
# if there is no value specified in head() or tail, it will return first 5 rows
df.tail()

Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

In [14]:
# info() to get an overview and metadata information of the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB


From the output, we will know:
- 169 entries of data
- 4 columns (Duration,pulse,maxpulse,calories)
- the last column (calories) has 164 non-null values, which means there are 5 rows of missing value




In [24]:
# remove rows which has empty cells
new_df=df.dropna()
new_df

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[164 rows x 4 columns]

In [25]:
print(new_df.to_string())

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45    105       132     246.0
24         60   

In [29]:
# dropna() methods will returns a new DataFrame and will not manipulate the original
# if to change to original, simply use 'inplace=True' argument
df.dropna(inplace = True)
df
# the output will be the same as before but this case we did in the original dataset

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[164 rows x 4 columns]

In [30]:
# there are other technique as well to replace null values, remove null entries isn't the best option
# we can fill it specifically using 'df.fillna({value],inplace=True})
df.fillna(130,inplace=True)

In [31]:
# replaced only for specified columns
df['Calories'].fillna(130,inplace=True)

In [33]:
# replaced using mean
x=df['Calories'].mean()
df['Calories'].fillna(x,inplace=True)

In [34]:
# replaced using median
x=df['Calories'].median()
df['Calories'].fillna(x,inplace=True)

In [35]:
# replaced using mode
x=df['Calories'].mode()[0]
df['Calories'].fillna(x,inplace=True)